tests/apm/monitoring-commandFailed-003.phpt

```diff
@@ -0,0 +1,60 @@
+--TEST--
+MongoDB\Driver\Monitoring\CommandFailedEvent
+--SKIPIF--
+<?php require __DIR__ . "/../utils/basic-skipif.inc"; ?>
+<?php skip_if_not_live(); ?>
+<?php skip_if_server_version('<', '3.4'); ?>
+--FILE--
+<?php
+require_once __DIR__ . "/../utils/basic.inc";
+
+$manager = new MongoDB\Driver\Manager(URI);
+
+class MySubscriber implements MongoDB\Driver\Monitoring\CommandSubscriber
+{
+    public function commandStarted( \MongoDB\Driver\Monitoring\CommandStartedEvent $event )
+    {
+        echo "started: ", $event->getCommandName(), "\n";
+    }
+
+    public function commandSucceeded( \MongoDB\Driver\Monitoring\CommandSucceededEvent $event )
+    {
+      var_dump($event);
+    }
+
+    public function commandFailed( \MongoDB\Driver\Monitoring\CommandFailedEvent $event )
+    {
+        echo "failed: ", $event->getCommandName(), "\n";
+        var_dump($event->getReply());
+    }
+}
+
+$subscriber = new MySubscriber;
+
+MongoDB\Driver\Monitoring\addSubscriber( $subscriber );
+
+$command = new MongoDB\Driver\Command([
+  'findAndModify' => COLLECTION_NAME,
+  'query' => ['_id' => 'foo'],
+  'upsert' => true,
+  'new' => true,
+]);
+
+try {
+    $manager->executeWriteCommand(DATABASE_NAME, $command);
+} catch (MongoDB\Driver\Exception\CommandException $e) {}
+
+?>
+--EXPECTF--
+started: findAndModify
+failed: findAndModify
+object(stdClass)#%d (4) {
+  ["ok"]=>
+  float(0)
+  ["errmsg"]=>
+  string(49) "Either an update or remove=true must be specified"
+  ["code"]=>
+  int(9)
+  ["codeName"]=>
+  string(13) "FailedToParse"
+}
```

Comment:
This test fails because `var_dump($event->getReply())` returns an `object` that includes `operationTime`, `$clusterTime`, and `signature`.

This can be fixed by adding
```
["operationTime"]=>
  object(MongoDB\BSON\Timestamp)#%d (2) {
    ["increment"]=>
    string(1) "1"
    ["timestamp"]=>
    string(10) "%d"
  }
  ["$clusterTime"]=>
  object(stdClass)#%d (2) {
    ["clusterTime"]=>
    object(MongoDB\BSON\Timestamp)#%d (2) {
      ["increment"]=>
      string(1) "1"
      ["timestamp"]=>
      string(10) "%d"
    }
    ["signature"]=>
    object(stdClass)#%d (2) {
      ["hash"]=>
      object(MongoDB\BSON\Binary)#%d (2) {
        ["data"]=>
        string(20) "%s"
        ["type"]=>
        int(0)
      }
      ["keyId"]=>
      int(0)
    }
```

Reply:
Thanks for catching that. It didn't come up with `overview.php` because I was already using `%A` there for 3.2 compatibility.